debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/rest/DebeziumMongoDbConnectorResourceIT.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.rest;
+
+import static io.debezium.testing.testcontainers.testhelper.RestExtensionTestInfrastructure.DATABASE;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.startsWith;
+
+import java.util.Locale;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.testcontainers.utility.MountableFile;
+
+import io.debezium.connector.mongodb.Module;
+import io.debezium.connector.mongodb.MongoDbConnector;
+import io.debezium.connector.mongodb.MongoDbConnectorConfig;
+import io.debezium.testing.testcontainers.ConnectorConfiguration;
+import io.debezium.testing.testcontainers.testhelper.RestExtensionTestInfrastructure;
+import io.debezium.testing.testcontainers.util.DockerUtils;
+import io.restassured.http.ContentType;
+
+public class DebeziumMongoDbConnectorResourceIT {
+
+    public static final MountableFile INIT_SCRIPT_RESOURCE = MountableFile.forClasspathResource("/initialize-mongo-single.js");
+    public static final String INIT_SCRIPT_PATH = "/docker-entrypoint-initdb.d/initialize-mongo-single.js";
+
+    @BeforeClass
+    public static void checkCondition() {
+        Assume.assumeThat("Skipping DebeziumMongoDbConnectorResourceIT tests when assembly profile is not active!",
+                System.getProperty("isAssemblyProfileActive", "false"),
+                is("true"));
+    }
+
+    @Before
+    public void start() {
+        RestExtensionTestInfrastructure.setupDebeziumContainer(Module.version(), DebeziumMongoDbConnectRestExtension.class.getName());
```

Comment:
Why not use `MongoDbDatabaseProvider`? That way we can assure that the initial container infrastructure is consistent across tests as is the imagehandling, port resolver (important for MacOS) and so on.

Reply:
Why didn't you do this in the Debezium when you was working with Anisha? This task was about moving what we have in the UI backend to connector-specific Connect REST extensions. If we can improve then contributions are welcome. ;)

Reply:
I agree with Jakub on this one. If you are adding new tests (whatever) tests, it should be consistent with the previous ones. The Mongo connector tests are the only tests that are stable and I am not willing to trade this because of this.

Reply:
1) I wasn't wroking with anisha. I was hotfixing a code which should have been gradually upgraded (and never was) and which was using MongoDB without the udnerstanding how the database networking works
2) This provider is meant for the mongo connector module tests -- though it acts as a factory for the varius configurations used by these tests it is not trying to contain configuration required by all other modules
3) Porting something from one codebase to another is not just a matter of simply moving the code.

Reply:
Filed https://issues.redhat.com/browse/DBZ-6946 to evaluate switching to `MongoDbDatabaseProvider` eventually.